src/CommandingOfficers/Venge.java

```diff
@@ -133,6 +134,12 @@ public void changeCombatContext(CombatContext instance)
 
         instance.defender = instance.attacker;
         instance.attacker = minion;
+
+        // Since we're swapping the combatants, we also need to swap the prediction polarity
+        if( instance.calcType == CalcType.PESSIMISTIC )
+          instance.calcType = CalcType.OPTIMISTIC;
+        else if( instance.calcType == CalcType.OPTIMISTIC )
+          instance.calcType = CalcType.PESSIMISTIC;
```

Comment:
Swapping attacker/defender is a common enough use-case - would it make more sense to create a `CombatContext` function that handles all this nonsense so that individual COs don't have to worry about it?

Reply:
I am mildly amused that we went from condemning this as weird jank to "a common usecase".
I do like the DRY you propose, though.

Reply:
I mean, it kinda is weird jank, but it's also cart-approved weird jank.